pkg/gardenlet/controller/backupbucket/reconciler.go

```diff
@@ -364,31 +362,31 @@ func (r *Reconciler) syncGeneratedSecretToGarden(ctx context.Context, backupBuck
 				Namespace: r.GardenNamespace,
 			},
 		}
-		ownerRef := metav1.NewControllerRef(backupBucket, gardencorev1beta1.SchemeGroupVersion.WithKind("BackupBucket"))
 
-		if _, err := controllerutils.GetAndCreateOrMergePatch(ctx, r.GardenClient, gardenGeneratedSecret, func() error {
-			gardenGeneratedSecret.OwnerReferences = []metav1.OwnerReference{*ownerRef}
-			controllerutil.AddFinalizer(gardenGeneratedSecret, finalizerName)
-			gardenGeneratedSecret.Data = seedGeneratedSecret.DeepCopy().Data
-			return nil
-		}); err != nil {
-			return err
-		}
-
-		gardenGeneratedSecretRef = &corev1.SecretReference{
+		// Update the BackupBucket status here before going for the CreateOrGetAndStrategicMergePatch call, so that the SeedAuthorizer
+		// can add the entry for this secret in the graph. See https://github.com/gardener/gardener/issues/7705 for more details.
+		patch := client.MergeFrom(backupBucket.DeepCopy())
+		backupBucket.Status.GeneratedSecretRef = &corev1.SecretReference{
 			Name:      gardenGeneratedSecret.Name,
 			Namespace: gardenGeneratedSecret.Namespace,
 		}
-	}
-
-	if gardenGeneratedSecretRef != nil || extensionBackupBucket.Status.ProviderStatus != nil {
```

Comment:
Don't we still need this

```golang
if extensionBackupBucket.Status.ProviderStatus != nil {
```

condition here?

Reply:
Good catch, adapted.